komfydb/storage/heap_page.cc

```diff
@@ -0,0 +1,122 @@
+#include "komfydb/storage/heap_page.h"
+#include "komfydb/common/int_field.h"
+#include "komfydb/common/string_field.h"
+#include "komfydb/config.h"
+#include "komfydb/utils/status_macros.h"
+
+namespace {
+using namespace komfydb::common;
+}
+
+namespace komfydb::storage {
+
+absl::StatusOr<std::unique_ptr<HeapPage>> HeapPage::Create(
+    PageId id, TupleDesc td, std::vector<uint8_t> data) {
```

Comment:
Change arguments to `TupleDesc* td, const std::vector<uint8_t>& data`. First change is because Tuples reference this td (they don't copy it), so if we pass TupleDesc by value then it will be freed by the end of this function. The second change is just not to copy the vector.